code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -288,30 +292,30 @@
 
 /obj/machinery/portable_atmospherics/canister/update_icon_state()
 	if(machine_stat & BROKEN)
-		icon_state = "[icon_state]-1"
+		icon_state = "[base_icon_state]-1"
+	else
+		base_icon_state = icon_state
```

Comment:
Are you sure you'd be changing the `base_icon_state` here instead of doing something like this?
`icon_state = "[base_icon_state]-[machine_stat & BROKEN]"`

Reply:
I'm using that as a place to cache the base_icon_state, because I don't want to clog up the defs of all the canisters with it.

Reply:
Interesting method

Reply:
"clogging defines" is not an issue and this is how you get awful code like turf base icon.

Reply:
Maybe try the method we are suggesting?

Reply:
Yep, I'll fix it up